### PR TITLE
[GEOT-6535] ImageWorker getHistogram return always the first histogra…

### PR DESCRIPTION
**Pull request by taba90**
…m computed

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer.

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):
- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by taba90** [comment]
Related Jira ticket: https://osgeo-org.atlassian.net/browse/GEOT-6535

File: modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java [hunk 1]
```diff
@@ -1137,75 +1137,83 @@ private Object getComputedProperty(final String name) {
...
-                HistogramWrapper wrapper =
-                        new HistogramWrapper(numBins, lowValues, highValues, bins);
-                // Setting the property
-                if (image instanceof PlanarImage) {
-                    ((PlanarImage) image).setProperty(HISTOGRAM, wrapper);
-                } else {
-                    PlanarImage p = getPlanarImage();
-                    p.setProperty(HISTOGRAM, wrapper);
-                    image = p;
+            // Image parameters
```
Comment (aaime) [inline modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java]: What happened here, lots of reformats?

File: modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java [hunk 2]
```diff
@@ -1137,75 +1137,83 @@ private Object getComputedProperty(final String name) {
...
-            pb.setSource(image, 0);
-            if (JAIExt.isJAIExtOperation("Stats")) {
-                StatsType[] stats = new StatsType[] {StatsType.HISTOGRAM};
-                // Band definition
-                int numBands = getNumBands();
-                int[] bands = new int[numBands];
-                for (int i = 0; i < numBands; i++) {
-                    bands[i] = i;
-                }
+        if (histogram instanceof HistogramWrapper) {
```
Comment (aaime) [inline modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java]:
```suggestion
        // can reuse cached histogram only if the bucket definitions are the same
        if (histogram instanceof HistogramWrapper) {
```

File: modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java (hunk 1 above)
Comment (aaime) [reply modules/library/coverage/src/main/java/org/geotools/image/ImageWorker.java]: Ah ok, saw it now, the caching check was inverted, which un-indented the block below one level.

**Comment by aaime** [comment]
Merged my comment suggestion, waiting for the build server to verify it. No further action required.

**Comment by aaime** [comment]
Merged, @taba90 if you want to backport it's good to go.

**Pull request merged**